### PR TITLE
Enhancement/8515 prevent ads module setup with ad block

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #8515

## Relevant technical choices

- Deviated from the IB slightly in that I extracted a new reusable component for the `AdBlockerWarningMessage` and used the existing `AdBlockerWarning` as a wrapper that can be duplicated. This prevents doing additional selector calls in the parent components of the AdBlockerWarning component to fetch the `getHelpLink` and `warningMessage` prop values.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/AdBlockerWarningMessage.js
Comment (tofumatt) [inline assets/js/components/AdBlockerWarningMessage.js:42]: What does this comment mean? When would `warningMessage` be loading? Given it's either `false` or a string here 🤔 

It looks like this was [copied from the original component](https://github.com/google/site-kit-wp/blob/5268002853636bb5a7fa5e01b38097f02cb3b67b/assets/js/modules/adsense/datastore/adblocker.js#L150-L154), where this was fetched from the `getAdBlockerWarningMessage`, but in that case we should adjust the prop types for this component, since it only says it expects a `string` but really that warning message is either a string or null.

`warningMessage` should probably be default `null` instead of `false`, to be inline with the types returned by the selector. Or even just `undefined` 🤷🏻

File: assets/js/components/AdBlockerWarningMessage.js
Comment (tofumatt) [inline assets/js/components/AdBlockerWarningMessage.js:76]: See above re: prop types here.

```suggestion
	warningMessage: PropTypes.oneOfType( [ PropTypes.null, PropTypes.string ] )
```

File: assets/js/googlesitekit/datastore/user/adblocker.js
Comment (tofumatt) [inline assets/js/googlesitekit/datastore/user/adblocker.js:84]: I know this is copied from before, but let's add some breathing room while we're here 😉 
```suggestion

		// The above is good about detecting most adblockers.
		// For the rest, we'll make a placeholder request to the favicon with some
		// additional stuff in the query string to (hopefully) trigger a filter.
		// If this throws, then the fetch request failed completely and we'll assume it was blocked.
		try {
			const params = [
				// The name of the parameter here doesn't really matter
				// since adblockers look at the URL as a whole.
				// Note: this value must not be URL-encoded.
				'google-site-kit=/adsense/pagead2.googlesyndication.com/pagead/js/adsbygoogle.js',
				// Add a timestamp for cache-busting.
				`timestamp=${ Date.now() }`,
			];
			await fetch( `/favicon.ico?${ params.join( '&' ) }`, {
				credentials: 'omit',
				// Don't follow any redirects; we only care about this request being blocked or not.
				redirect: 'manual',
			} );
		} catch {
			return true;
		}

		return false;
```

File: assets/js/googlesitekit/datastore/user/adblocker.js
Comment (tofumatt) [inline assets/js/googlesitekit/datastore/user/adblocker.js:113]: I suspect "checking the global" is outdated here regarding what we check, given this [original commit is four years old](https://github.com/google/site-kit-wp/commit/f4263c27379eb19212e057fc9c631dd448e6ed54) 😅 

```suggestion
		// If the ad blocker status is already set, don't make any requests
		// to check the ad blocker status.
```

File: assets/js/googlesitekit/datastore/user/adblocker.js
Comment (tofumatt) [inline assets/js/googlesitekit/datastore/user/adblocker.js:128]:
```suggestion
	 * @since 1.9.0
	 * @since n.e.x.t Selector moved from the `MODULES_ADSENSE` store to the `CORE_USER` store.
```

File: assets/js/components/AdBlockerWarningMessage.js
Comment (tofumatt) [inline assets/js/components/AdBlockerWarningMessage.js:38]: What is this prop for? I can't find any instance of it actually being used or called 🤔

File: assets/js/modules/ads/components/common/AdBlockerWarning.js
Comment (tofumatt) [inline assets/js/modules/ads/components/common/AdBlockerWarning.js:34]: I can't find anywhere this prop is actually used, should we remove it for now?

Screenshot 2024-04-16 at 21 24 03

File: assets/js/modules/ads/components/common/AdBlockerWarning.js
Comment (tofumatt) [inline assets/js/modules/ads/components/common/AdBlockerWarning.js:46]: As mentioned above, `AdBlockerWarningMessage` needs its prop types adjusting as this will pass `null`, not `false`.

File: assets/js/modules/ads/components/setup/SetupMain.js
Comment (tofumatt) [inline assets/js/modules/ads/components/setup/SetupMain.js:65]: I wonder if, technically, we should be using `%s` for "Google Ads" with an `_x( 'Google Ads', 'Service name', 'google-site-kit' )` here 🤔

File: assets/js/modules/adsense/components/common/AdBlockerWarning.js
Comment (tofumatt) [inline assets/js/modules/adsense/components/common/AdBlockerWarning.js:34]: I think we can remove this prop too, it's totally unused.

File: tests/backstop/reference/google-site-kit_Modules_Ads_Setup_SetupForm_Empty_0_document_0_small.png
Comment (tofumatt) [inline tests/backstop/reference/google-site-kit_Modules_Ads_Setup_SetupForm_Empty_0_document_0_small.png:1]: Why the addition of the header separator only for the Ads module? I don't see it anywhere else in the plugin:

Screenshot 2024-04-16 at 21 48 19
Screenshot 2024-04-16 at 21 39 02

If it was in a Figma design we should update it across the plugin in a separate issue rather than piecemeal across modules 🤔

**Review by tofumatt — CHANGES_REQUESTED** [review]
Largely looks good, I think this is a good refactor of where the "is there an ad blocker?" logic is housed.

There are some issues with the components prop types and such, along with a design issue I mentioned around one of the VRT image changes. After that I think this is good-to-go 👍🏻

File: assets/js/components/AdBlockerWarningMessage.js
Comment (jimmymadon) [reply assets/js/components/AdBlockerWarningMessage.js:38]: This prop comes down from the parent `<AdBlockerWarning>` component. However, you are right, every time this componet is used, we have never passed in a context now. There are scss selectors for different contexts though, so should we keep this or just remove the prop and the selectors. Not sure if this is dynamically called anywhere else? 

Screenshot 2024-04-17 at 13 32 36

File: assets/js/modules/ads/components/setup/SetupMain.js
Comment (jimmymadon) [reply assets/js/modules/ads/components/setup/SetupMain.js:65]: This code is actually part of another PR - it shows as an addition due to perhaps a merge conflict resolution. Doing a quick search, we've always used the service name without the Google prefix. I don't want to change the wording without approval and in the sense of time, should we deal with this separately?

File: assets/js/components/AdBlockerWarningMessage.js
Comment (jimmymadon) [reply assets/js/components/AdBlockerWarningMessage.js]: Well, based on your comments below, I have deleted the prop. I'm tempted to delete the unused scss selectors too but one of them (modules-list) is actually being used. So will keep these for now.

Comment (jimmymadon) [reply assets/js/components/AdBlockerWarningMessage.js]: I actually found where this prop _should_ have been used! The `DefaultSettingsIncomplete` component uses the `ModuleSettingsWarning` component. However, AdSense uses its own `SettingsSetupIncomplete` component and renders the AdBlockerWarning:

https://github.com/google/site-kit-wp/blob/8953c3475a1a22c98c9dea9bc4d3a7a115bfd112/assets/js/modules/adsense/components/settings/SettingsSetupIncomplete.js#L60

For consistency, we should add the `context` to `AdBlockerWarning` to be `modules-list`. However again, I actually don't like the design of the modules list warning which is smaller and looks broken! 

This looks broken:
Screenshot 2024-04-17 at 15 13 15


Compared to this which looks better IMO:
Screenshot 2024-04-17 at 15 13 26

Should I create a new issue referencing this comment to get UX to redesign the warning? Let's not hold up this PR.

Comment (tofumatt) [reply assets/js/components/AdBlockerWarningMessage.js]: I'd be happy to just remove the CSS there and have everything default to the larger size, but sure, let's do it in a new issue. 👍🏻

File: assets/js/modules/ads/components/setup/SetupMain.js
Comment (tofumatt) [reply assets/js/modules/ads/components/setup/SetupMain.js:65]: Yeah let's do it separately, all good 👍🏻

File: tests/backstop/reference/google-site-kit_Modules_Ads_Setup_SetupForm_Empty_0_document_0_small.png
Comment (tofumatt) [reply tests/backstop/reference/google-site-kit_Modules_Ads_Setup_SetupForm_Empty_0_document_0_small.png:1]: Oh, yeah, that's a good call. Let's do that and the improved adblocker warning message all as another issue 👍🏻

**Pull request merged**
